src/libraries/System.Formats.Tar/tests/TarFile/TarFile.ExtractToDirectoryAsync.File.Tests.cs

```diff
@@ -223,29 +225,38 @@ public async Task UnixFileModes_Async()
                 writer.WriteEntry(outOfOrderDir);
             }
 
-            await TarFile.ExtractToDirectoryAsync(archivePath, destination.Path, overwriteFiles: false);
-
             string dirPath = Path.Join(destination.Path, "dir");
-            Assert.True(Directory.Exists(dirPath), $"{dirPath}' does not exist.");
+            string filePath = Path.Join(destination.Path, "file");
+            string missingParentPath = Path.Join(destination.Path, "missing_parent");
+            string missingParentDirPath = Path.Join(missingParentPath, "dir");
+            string outOfOrderDirPath = Path.Join(destination.Path, "out_of_order_parent");
+
+            if (overwrite)
+            {
+                File.OpenWrite(filePath).Dispose();
+                Directory.CreateDirectory(dirPath);
+                Directory.CreateDirectory(missingParentDirPath);
+                Directory.CreateDirectory(outOfOrderDirPath);
+            }
+
+            await TarFile.ExtractToDirectoryAsync(archivePath, destination.Path, overwriteFiles: overwrite);
+
+           Assert.True(Directory.Exists(dirPath), $"{dirPath}' does not exist.");
             AssertFileModeEquals(dirPath, TestPermission1);
 
-            string filePath = Path.Join(destination.Path, "file");
             Assert.True(File.Exists(filePath), $"{filePath}' does not exist.");
             AssertFileModeEquals(filePath, TestPermission2);
 
-            // Missing parents are created with DefaultDirectoryMode.
-            string missingParentPath = Path.Join(destination.Path, "missing_parent");
+            // Missing parents are created with CreateDirectoryDefaultMode.
             Assert.True(Directory.Exists(missingParentPath), $"{missingParentPath}' does not exist.");
-            AssertFileModeEquals(missingParentPath, DefaultDirectoryMode);
+            AssertFileModeEquals(missingParentPath, CreateDirectoryDefaultMode);
 
-            string missingParentDirPath = Path.Join(missingParentPath, "dir");
             Assert.True(Directory.Exists(missingParentDirPath), $"{missingParentDirPath}' does not exist.");
             AssertFileModeEquals(missingParentDirPath, TestPermission3);
 
             // Directory modes that are out-of-order are still applied.
-            string outOfOrderDirPath = Path.Join(destination.Path, "out_of_order_parent");
             Assert.True(Directory.Exists(outOfOrderDirPath), $"{outOfOrderDirPath}' does not exist.");
-            AssertFileModeEquals(outOfOrderDirPath, TestPermission4);
+            AssertFileModeEquals(outOfOrderDirPath, overwrite ? TestPermission4 : CreateDirectoryDefaultMode);
```

Comment:
If overwrite is false and we have an out-of-order directory, we don't honor the `Mode`, even though the directory is a product of our API? 

I think you should chmod the out-of-order directory if it was created by `TarFile.ExtractToDirectory` and then this should be: 
```suggestion
            AssertFileModeEquals(outOfOrderDirPath, TestPermission4);
```

Based on https://github.com/dotnet/runtime/pull/72078#discussion_r921432953, that's how tar do it, right?

Reply:
This behaves similar to tar.

If I add the `--no-overwrite-dir` option to not overwrite metadata, the `strace` becomes:
```
dir
mkdirat(AT_FDCWD, "dir", 0740)          = 0
file
missing_parent/dir
mkdirat(AT_FDCWD, "missing_parent/dir", 0704) = -1 ENOENT (No such file or directory)
mkdirat(AT_FDCWD, "missing_parent", 0755) = 0
mkdirat(AT_FDCWD, "missing_parent/dir", 0704) = 0
out_of_order_parent/file
mkdirat(AT_FDCWD, "out_of_order_parent", 0755) = 0
out_of_order_parent
mkdirat(AT_FDCWD, "out_of_order_parent", 0701) = -1 EEXIST (File exists)
```